src/createServer.js

```diff
@@ -1,8 +1,108 @@
 'use strict';
 
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+const mime = require('mime-types');
+const zlib = require('zlib');
+const { formidable } = require('formidable');
+const { pipeline } = require('stream');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', (req, res) => {
+    const url = new URL(req.url || '', `http://${req.headers.host}`);
+    const requestedPath = url.pathname.slice(1) || 'index.html';
+
+    if (requestedPath === 'compress' && req.method === 'GET') {
+      res.writeHead(400, { 'content-type': 'text/plain' });
+      res.end('Request should be POST method!');
+
+      return;
+    }
+
+    if (requestedPath === 'compress' && req.method === 'POST') {
+      const form = formidable({});
+
+      const compressors = {
+        gzip: zlib.createGzip,
+        br: zlib.createBrotliCompress,
+        deflate: zlib.createDeflate,
+      };
+
+      form.parse(req, (err, { compressionType }, { file }) => {
+        if (err || !file || !compressionType) {
+          res.statusCode = 400;
+          res.end('Form error');
+
+          return;
+        }
+
+        if (!compressors.hasOwnProperty(compressionType[0])) {
```

Comment:
The 'compressionType' field is assumed to be an array. Ensure that it is validated properly before accessing its first element. Consider adding a check to confirm that 'compressionType' is an array and has at least one element.